src/DDPandoraPFANewProcessor.cc

```diff
@@ -159,24 +160,40 @@ void DDPandoraPFANewProcessor::init()
         this->FinaliseSteeringParameters();
 
         m_pPandora = new pandora::Pandora();
-        m_pGeometryCreator = new DDGeometryCreator(m_geometryCreatorSettings, m_pPandora);
-        m_pCaloHitCreator = new DDCaloHitCreator(m_caloHitCreatorSettings, m_pPandora);
-        
-        
+        if(m_settings.m_detectorName == "ALLEGRO")
+        {
+          m_settings.m_trackCreatorName = "DDTrackCreatorALLEGRO";
+          m_pGeometryCreator = new DDGeometryCreatorALLEGRO(m_geometryCreatorSettings, m_pPandora);
+          m_pCaloHitCreator = new DDCaloHitCreatorALLEGRO(m_caloHitCreatorSettings, m_pPandora);
+        }
+        else
+        {
+          m_pGeometryCreator = new DDGeometryCreator(m_geometryCreatorSettings, m_pPandora);
+          m_pCaloHitCreator = new DDCaloHitCreator(m_caloHitCreatorSettings, m_pPandora);
+        }
+
         ///FIXME: IMPLEMENT FACTORY
         if (m_settings.m_trackCreatorName == "DDTrackCreatorCLIC")
             m_pTrackCreator = new DDTrackCreatorCLIC(m_trackCreatorSettings, m_pPandora);
         else if (m_settings.m_trackCreatorName == "DDTrackCreatorILD")
             m_pTrackCreator = new DDTrackCreatorILD(m_trackCreatorSettings, m_pPandora);
+        else if (m_settings.m_trackCreatorName == "DDTrackCreatorALLEGRO")
+            m_pTrackCreator = new DDTrackCreatorALLEGRO(m_trackCreatorSettings, m_pPandora);
         else
             streamlog_out(ERROR) << "Unknown DDTrackCreator: "<<m_settings.m_trackCreatorName << std::endl;
 
-
         m_pDDMCParticleCreator = new DDMCParticleCreator(m_mcParticleCreatorSettings, m_pPandora);
         m_pDDPfoCreator = new DDPfoCreator(m_pfoCreatorSettings, m_pPandora);
 
         PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, this->RegisterUserComponents());
-        PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, m_pGeometryCreator->CreateGeometry());
+        if(m_settings.m_detectorName == "ALLEGRO")
+        {
+          PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, static_cast<DDGeometryCreatorALLEGRO*>(m_pGeometryCreator)->CreateGeometry());
+        }
+        else
+        {
+          PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, m_pGeometryCreator->CreateGeometry());
+        }
```

Comment:
```suggestion
        PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, m_pGeometryCreator->CreateGeometry());
```
This isn't necessary anymore with the inheritance and virtual functions.